### PR TITLE
feat: added metrics for member and replica-set avg health of MongoDB …

### LINKED ISSUE #11515: inputs.mongodb : adding metrics for member and replica-set avg health
## Feature Request

The MongoDB input plugin offers currently no metrics to detect the health state of a replica-set.

### Proposal:
Collecting MongoDB replica-set health metrics

### Current behavior:
No MongoDB replica-set health metrics available

### Desired behavior:

- Collecting a replica-set health metric of the MongoDB cluster node your Telegraf agent is connected to
- Collecting a replica-set health average metric, to detect if another member is unhealthy from the perspective of another member and also to see how bad the replica-set health is

### Use case:
If Telegraf would offer this mentioned MongoDB replica-set health metrics, you would get everything needed to build monitoring alerts on top of this data.

### PR DESCRIPTION
**Pull request by svenburkard**
…input plugin

- [x]  make check 
- [x]  make check-deps
- [x]  make test
- [x]  make docs
- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves #11515

Added metrics for member and replica-set average health of MongoDB input plugin.
With this additional Telegraf metrics, you have all needed data ready to build monitoring alerts on unhealthy MongoDB replica-set events.

### CODE REVIEW

File: plugins/inputs/mongodb/mongostat.go [hunk 1]
```diff
@@ -1356,6 +1361,26 @@ func NewStatLine(oldMongo, newMongo MongoStatus, key string, all bool, sampleSec
 					returnVal.ReplLag = lag
 				}
 			}
+
+			// Prepartions for the average health state of the replica-set
+			var replMemberCount int = len(newReplStat.Members)
+			var replMemberHealthyCount int = 0
```
Comment (srebhan) [inline plugins/inputs/mongodb/mongostat.go:1367]: Please use
```suggestion
			replMemberCount := len(newReplStat.Members)
			replMemberHealthyCount := 0
```

to make the linter happy and make the code look nicer. ;-)

File: plugins/inputs/nats_consumer/README.md [hunk 2]
```diff
@@ -24,7 +24,7 @@ instances of telegraf can read from a NATS cluster in parallel.
   ## With jetstream the nats-server persists messages and 
   ## a consumer can consume historical messages. This is
   ## useful when telegraf needs to restart it don't miss a 
-  ## message. You need to configure the nats-server:
+  ## message. You need to configure the nats-server.
```
[the remark this answers is not in the document]
Comment (powersj) [reply plugins/inputs/nats_consumer/README.md:27]: This is from running `make docs` yesterday after we merged in #11373. It broke master, so I created and merged #11514.

**Comment by svenburkard** [comment]
Hi @srebhan, thanks for your feedback. The suggested changes have been added with a fresh rebase of the current master and the Lint CI job seems to be happy now :)

File: plugins/inputs/mongodb/mongostat.go (hunk 1 above)
Comment (svenburkard) [reply plugins/inputs/mongodb/mongostat.go]: changed with e07ef0a1e

File: plugins/inputs/nats_consumer/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (svenburkard) [reply plugins/inputs/nats_consumer/README.md]: Yes, the CI job complaint yesterday i should do another `make docs`, which then introduced the changes above. Thanks for clarification. A fresh rebase of current master with the PR related changes has been done.

**Review by srebhan — APPROVED** [review]
Looks good to me. Thanks for the contribution @svenburkard!

**Comment by svenburkard** [comment]
Nice, you are welcome!

**Review by powersj — APPROVED** [review]

**Pull request merged**